sdk/trace/trace_test.go

```diff
@@ -1976,3 +1976,81 @@ func TestEmptyRecordingSpanAttributes(t *testing.T) {
 func TestEmptyRecordingSpanDroppedAttributes(t *testing.T) {
 	assert.Equal(t, 0, (&recordingSpan{}).DroppedAttributes())
 }
+
+func TestAddLinkWithInvalidSpanContext(t *testing.T) {
+	te := NewTestExporter()
+	sl := NewSpanLimits()
+	tp := NewTracerProvider(
+		WithSpanLimits(sl),
+		WithSyncer(te),
+		WithResource(resource.Empty()),
+	)
+	span := startSpan(tp, "AddSpanWithInvalidSpanContext")
+	inValidContext := trace.NewSpanContext(trace.SpanContextConfig{
+		TraceID: trace.TraceID([16]byte{}),
+		SpanID:  [8]byte{},
+	})
+	attrs := []attribute.KeyValue{{Key: "k", Value: attribute.StringValue("v")}}
+	span.AddLink(trace.Link{
+		SpanContext: inValidContext,
+		Attributes:  attrs,
+	})
+
+	want := &snapshot{
+		name: "span0",
+		spanContext: trace.NewSpanContext(trace.SpanContextConfig{
+			TraceID:    tid,
+			TraceFlags: 0x1,
+		}),
+		parent:               sc.WithRemote(true),
+		links:                nil,
+		spanKind:             trace.SpanKindInternal,
+		instrumentationScope: instrumentation.Scope{Name: "AddSpanWithInvalidSpanContext"},
+	}
+	got, err := endSpan(te, span)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if diff := cmpDiff(got, want); diff != "" {
+		t.Errorf("AddLinkWithInvalidSpanContext: -got +want %s", diff)
+	}
+}
+
+func TestAddLink(t *testing.T) {
```

Comment:
Both this test and the one above are very similar. How about refactoring them into a single one with options, and use subtests?
https://go.dev/blog/subtests

Reply:
Extracted into #5114.